test/utils/behaviours.ts

```diff
@@ -207,23 +209,42 @@ export const txShouldSetVariableAndEmitEvent = async ({
   });
 };
 
-export const fnShouldOnlyBeCallableByGovernance = (
-  delayedContract: () => Contract,
-  fnName: string,
-  governance: Impersonator,
-  args: unknown[] | (() => unknown[])
-): void => {
-  it('should be callable by governance', () => {
-    return expect(callFunction(governance)).not.to.be.revertedWith('OnlyGovernance()');
+export const shouldBeExecutableOnlyByGovernor = ({
+  contract,
+  funcAndSignature,
+  params,
+  governor,
+}: {
+  contract: () => Contract;
+  funcAndSignature: string;
+  params?: any[] | (() => any[]);
+  governor: () => SignerWithAddress | Wallet;
+}) => {
+  let realParams: any[];
+  given(() => {
+    realParams = typeof params === 'function' ? params() : params ?? [];
   });
-
-  it('should not be callable by any address', async () => {
-    return expect(callFunction(await wallet.generateRandom())).to.be.revertedWith('OnlyGovernance()');
+  when('not called from governor', () => {
+    let onlyGovernorAllowedTx: Promise<TransactionResponse>;
+    given(async () => {
+      const notGovernor = await wallet.generateRandom();
+      onlyGovernorAllowedTx = contract()
+        .connect(notGovernor)
+        [funcAndSignature](...realParams!);
+    });
+    then('tx is reverted with reason', async () => {
+      await expect(onlyGovernorAllowedTx).to.be.revertedWithCustomError(contract(), 'OnlyGovernor');
```

Comment:
Had to start using `revertedWithCustomError` post `hardhat 2.10` and `@nomicfoundation/hardhat-toolbox`.